src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java

```diff
@@ -748,20 +749,20 @@ public BuildStepMonitor getRequiredMonitorService() {
      * behave sensibly when a later build actually finishes before an earlier
      * one.
      *
-     * @param build a build for which we may be sending mail
+     * @param run a build for which we may be sending mail
      * @param listener a listener to which we may print warnings in case the
      * actual previous build is still in progress
      * @return the previous build, or null if that build is missing, or is still
      * in progress
      */
     public static @CheckForNull
-    AbstractBuild<?, ?> getPreviousBuild(@Nonnull AbstractBuild<?, ?> build, TaskListener listener) {
-        AbstractBuild<?, ?> previousBuild = build.getPreviousBuild();
-        if (previousBuild != null && previousBuild.isBuilding()) {
-            listener.getLogger().println(Messages.ExtendedEmailPublisher__is_still_in_progress_ignoring_for_purpo(previousBuild.getDisplayName()));
+    Run<?, ?> getPreviousRun(@Nonnull Run<?, ?> run, TaskListener listener) {
```

Comment:
I don't know if any other plugin (that depends on this one) is using this public method. If so, this change will break binary compatibility. If you want to fix it you can use `@WithBridgeMethods` ([in this way](https://github.com/jenkinsci/analysis-core-plugin/blob/42c50ab8f5cca9da0d83519a03afd7fe2a46de57/src/main/java/hudson/plugins/analysis/views/ConsoleDetail.java#L114))